casanovo/denovo/model.py

```diff
@@ -444,6 +461,37 @@ def _terminate_finished_beams(
                             abs(d) < self.precursor_mass_tol
                             for d in delta_mass_ppm
                         )
+
+                        # Don't terminate if m/z difference is by NH3 loss
+                        if (
+                            exceeds_precursor_mz_tol
+                            and not is_within_precursor_mz_tol
+                            and is_nh3_loss_in_vocab
+                        ):
+                            alt_peptide_seq = peptide_seq + ["-17.027"]
```

Comment:
**issue:** NH3 loss is an N-terminal modification, whereas here it's added at the end (i.e. C-terminal). So either the modification appears in the wrong location in the peptide string, or the peptide strings hasn't been reversed yet?
For mass calculation I guess the position doesn't really matter, but at the very least it's confusing. Worst case scenario it results in hard-to-diagnose bugs now or in the future.

Reply:
Peptide wasn't reversed at that point. Changed to this have peptide reversed before hand and modification added to N-terminus. Can revisit after `detokenize` resolved